### DISCUSSION TITLE
Vercel deployment error: `GLIBC_2.28' not found

### DISCUSSION THREAD

**Discussion by simon-bonnedahl** [vercel/next.js#44404] [Help]
Hey, I cant figure out what causes this issue. Can anyone help me?

Deployment logs:

<img width="1194" alt="SCR-20221228-s9w" src="https://user-images.githubusercontent.com/32783686/209862109-0639693c-65e4-4556-9cc5-62dd0915694f.png">

**Comment by icyJoseph** [comment]
Do you have a direct dependency to `canvas`? The NPM package? It looks like they do not support Node 18, for... reasons. Apparently locking it to 2.7.0, should do the trick. I got that from:

- https://github.com/Automattic/node-canvas/issues/1805
- https://github.com/Automattic/node-canvas/issues/2142


Does that apply to your case? I am a bit short on time to try and reproduce this myself, let me know how it goes.

**Reply by simon-bonnedahl** [reply]
Thank you for the suggestion. I think it does, the problem is now gone after changing the version of node in Vercel settings to 14.x.

**Reply by icyJoseph** [reply]
Nice! Don't forget to mark as answered.

Though, this means that you are blocked from upgrading to 16, until further notice. That's not very good.

**Reply by simon-bonnedahl** [reply]
Yeah, I did not like that fix. Do you have any other possible solution?

**Comment by mino9421** [comment]
I fixed this by switching to node 20 from vercel settings and adding the following config top of the next.config file
`if (
  process.env.LD_LIBRARY_PATH == null ||
  !process.env.LD_LIBRARY_PATH.includes(
    `${process.env.PWD}/node_modules/canvas/build/Release:`
  )
) {
  process.env.LD_LIBRARY_PATH = `${
    process.env.PWD
  }/node_modules/canvas/build/Release:${process.env.LD_LIBRARY_PATH || ""}`;
}`

also my canvas is 2.8.0

if those 3 changes does not help please let me know im jobless and free to help if you get me a reference 🗡️
